modules/apps/depot/depot-web/src/main/java/com/liferay/depot/web/internal/template/StagingIndicatorTemplateContextContributor.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.depot.web.internal.template;
+
+import com.liferay.depot.web.internal.util.StagingIndicatorUtil;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.service.permission.PortletPermission;
+import com.liferay.portal.kernel.template.TemplateContextContributor;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Evan Thibodeau
+ */
+@Component(
+	property = "type=" + TemplateContextContributor.TYPE_THEME,
+	service = TemplateContextContributor.class
+)
+public class StagingIndicatorTemplateContextContributor
+	implements TemplateContextContributor {
+
+	@Override
+	public void prepare(
+		Map<String, Object> contextObjects,
+		HttpServletRequest httpServletRequest) {
+
+		try {
+			if (StagingIndicatorUtil.isShowStagingIndicator(
+					httpServletRequest, _portletPermission)) {
+
+				StringBundler sb = new StringBundler(2);
+
+				sb.append(
+					GetterUtil.getString(contextObjects.get("bodyCssClass")));
+
+				sb.append(" has-staging-indicator");
+
+				contextObjects.put("bodyCssClass", sb.toString());
```

Comment:
A very minor thing: can you use the `concat` method so that everything can be inlined?
```
if (StagingIndicatorUtil.isShowStagingIndicator(
			httpServletRequest, _portletPermission)) {

		contextObjects.put(
			"bodyCssClass",
 				StringBundler.concat(
					GetterUtil.getString(
						contextObjects.get("bodyCssClass"))),
					 " has-staging-indicator");
}
```

Reply:
Other than this, everything looks good.

Reply:
Thanks @adolfopa . I updated with what looks to be the common pattern like here: https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/template/ChangeTrackingTemplateContextContributor.java#L45